File: gcd/go/gcd.go
package main

import "fmt"

func main() {
	m, n := 18, 30
	fmt.Printf("%d と %d の最大公約数は %d\n", m, n, gcd(m, n))
}

func gcd(m, n int) int {
	if n == 0 {
		return m
	}
	return gcd(n, m%n)
}

File: gcd/rust/gcd.rs
fn main() {
    let m = 18;
    let n = 30;
    println!("{} と {} の最大公約数は {}", m, n, gcd(m, n));
}

fn gcd(m: i32, n: i32) -> i32 {
    if n == 0 {
        return m;
    }
    return gcd(n, m % n);
}
